src/app/shell/admin-tools/platform/create-direction/create-direction.component.ts

```diff
@@ -1,50 +1,131 @@
+
 
-  constructor(
+    isActiveDepartmentInfoButton = false;
+    isActiveDirectionInfoButton = false;
+    isActiveClassInfoButton = false;
+
+    departmentFormGroup: FormGroup;
+    directionFormGroup: FormGroup;
+    ClassFormGroup: FormGroup;
+
+   constructor(
     private fb: FormBuilder,
-    private store: Store, 
-    private navigationBarService: NavigationBarService) { 
-      
-      this.directionFormGroup = this.fb.group({
-        image: new FormControl(''),
-        directionName: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)]),
-        sectionName: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)]),
-        className: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)]),
+    private store: Store,
+    private route: ActivatedRoute,
+    private matDialog: MatDialog,) { }
+
+    ngOnInit(): void {
+      this.ClassFormArray.push(this.newForm());
+      this.direction$.pipe(takeUntil(this.destroy$),filter((direction: Direction)=>!!direction)).subscribe((direction: Direction)=>this.direction = direction);
+      this.department$.pipe(takeUntil(this.destroy$),filter((department: Department)=>!!department)).subscribe((department: Department)=>this.department = department);
+    }
+
+    private newForm( ): FormGroup {
+      const ClassFormGroup = this.fb.group({
+        title: new FormControl('', [Validators.required, Validators.pattern(TEXT_REGEX)]),
+      });
+      return ClassFormGroup;
+    }
+
+  ngAfterViewInit(): void {
+    if (this.editMode) {
+      this.route.params.subscribe((params: Params) => {
+        this.stepper.selectedIndex = +createDirectionSteps[params.param];
       });
     }
+    }
+
+  addClass(): void {
+    this.ClassFormArray.push(this.newForm());
+  }
 
-  ngOnInit(): void {
-    this.store.dispatch(new AddNavPath(this.navigationBarService.creatNavPaths(
-      { name: NavBarName.AdminTools, isActive: false, disable: false },
-      { name: NavBarName.Platform, isActive: false, disable: false },
-      { name: NavBarName.Direction, isActive: false, disable: true }
-    )));
+  onDeleteForm(indexClass: number): void {
+    this.ClassFormArray.removeAt(indexClass);
   }
 
-  onSubmit(): void { }
+  onSubmit(): void {
+    if (this.ClassFormArray.invalid) {
+      this.checkValidationClass();
+    } else {
+      const dialogRef = this.matDialog.open(ConfirmationModalWindowComponent, {
+        width: '330px',
+        data: {
+          type: ModalConfirmationType.createClass,
+        }
+      });
+      dialogRef.afterClosed().subscribe((result: boolean) => {
+      const classes: IClass[] = [];
+      const department = this.store.selectSnapshot<Department>(AdminState.department);
+
+      this.ClassFormArray.controls.forEach((form: FormGroup) =>
+        classes.push(new IClass(form.value, department.id))
+        );
+        result && this.store.dispatch(new CreateClass(classes));
+      });
```

Comment:
firstly check for positive result, then do all the following actions 
otherwise we do classes creation but don't dispatch action